### ISSUE TITLE
Unexpected accuracy for float32

### ISSUE THREAD

**Issue by Oceania2018** [SciSharp/TensorFlow.NET#525]
When running this UnitTest: [tf.sin](https://github.com/SciSharp/TensorFlow.NET/blob/master/test/TensorFlowNET.UnitTest/math_test/MathOperationTest.cs), tf.net losts accuracy. This problem will lead to the vanishing of the gradient in the case of multilayer neural networks, especially more than 8 layers deep.

![image](https://user-images.githubusercontent.com/1705364/76690509-8e3a9280-660e-11ea-8bd9-4c7201253899.png)

Same unexpected result for [Sigmoid](https://github.com/SciSharp/TensorFlow.NET/blob/master/test/TensorFlowNET.UnitTest/nn_test/ActivationFunctionTest.cs) test.

What's wrong with it?
![image](https://user-images.githubusercontent.com/1705364/76691994-9a2e5080-661e-11ea-9c21-4a43c509bcda.png)

**Comment by Oceania2018** [comment]
`float` in python is not equal `float32` in .net.
